examples/mixed-multigrid-solver/mixed-multigrid-solver.cpp

```diff
@@ -130,40 +133,67 @@ int main(int argc, char* argv[])
         ir2::build()
             .with_solver(bj2::build().with_max_block_size(1u).on(exec))
-            .with_relaxation_factor(0.9f)
+            .with_relaxation_factor(static_cast<ValueType2>(0.9))
             .with_criteria(
-                gko::stop::Iteration::build().with_max_iters(2u).on(exec))
+                gko::stop::Iteration::build().with_max_iters(1u).on(exec))
             .on(exec));
     // Create RestrictProlong factory
     auto mg_level_gen =
-        gko::share(pgm::build().with_deterministic(true).on(exec));
+        gko::share(pgm::build().with_deterministic(false).on(exec));
     auto mg_level_gen2 =
-        gko::share(pgm2::build().with_deterministic(true).on(exec));
+        gko::share(pgm2::build().with_deterministic(false).on(exec));
     // Create CoarsesSolver factory
-    auto coarsest_solver_gen =
-        cg::build()
+    auto coarsest_solver_gen = gko::share(
+        ir::build()
+            .with_solver(bj::build().with_max_block_size(1u).on(exec))
+            .with_relaxation_factor(static_cast<ValueType>(0.9))
             .with_criteria(
                 gko::stop::Iteration::build().with_max_iters(4u).on(exec))
-            .on(exec);
+            .on(exec));
+    auto coarsest_solver_gen2 = gko::share(
+        ir2::build()
+            .with_solver(bj2::build().with_max_block_size(1u).on(exec))
+            .with_relaxation_factor(static_cast<ValueType2>(0.9))
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(4u).on(exec))
+            .on(exec));
     // Create multigrid factory
-    auto multigrid_gen = mg::build()
-                             .with_max_levels(2u)
-                             .with_min_coarse_rows(5u)
-                             .with_pre_smoother(smoother_gen, smoother_gen2)
-                             .with_post_uses_pre(true)
-                             .with_mg_level(mg_level_gen, mg_level_gen2)
-                             .with_coarsest_solver(gko::share(
-                                 bj2::build().with_max_block_size(1u).on(exec)))
-                             .with_criteria(iter_stop, tol_stop)
-                             .on(exec);
-
+    std::shared_ptr<gko::LinOpFactory> multigrid_gen;
+    if (use_mixed) {
+        std::cout << "USE" << std::endl;
+        multigrid_gen =
+            mg::build()
+                .with_max_levels(10u)
+                .with_min_coarse_rows(2u)
+                .with_pre_smoother(smoother_gen, smoother_gen2)
+                .with_post_uses_pre(true)
+                .with_mg_level(mg_level_gen, mg_level_gen2)
+                .with_level_selector([](const gko::size_type level,
+                                        const gko::LinOp*) -> gko::size_type {
+                    std::cout << "level " << level << std::endl;
+                    return level >= 1 ? 1 : 0;
+                })
+                .with_coarsest_solver(coarsest_solver_gen2)
```

Comment:
I think you should add some comments to explain to the user what's the difference compared to the normal non-mixed mode. Either in this example or the other one.